pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1113,7 +1113,11 @@ public void addTable(@Nonnull TableConfig tableConfig)
         ZKMetadataProvider.setRealtimeTableConfig(_propertyStore, tableNameWithType, tableConfig.toZNRecord());
 
         // Update replica group partition assignment to the property store if applicable
-        updateReplicaGroupPartitionAssignment(tableConfig);
+        IndexingConfig indexingConfig = tableConfig.getIndexingConfig();
+        StreamConfig streamConfig = new StreamConfig(indexingConfig.getStreamConfigs());
+        if (streamConfig.hasLowLevelConsumerType() && !streamConfig.hasHighLevelConsumerType()) {
```

Comment:
So, while migrating we cannot use replica based segment assignment? Not sure why, but it is good to add some comments here as to why this check is being made.